torch/nn/modules/rnn.py

```diff
@@ -41,14 +45,22 @@ def __init__(self, mode, input_size, hidden_size,
                 suffix = '_reverse' if direction == 1 else ''
                 weights = ['weight_ih_l{}{}', 'weight_hh_l{}{}', 'bias_ih_l{}{}', 'bias_hh_l{}{}']
                 weights = [x.format(layer, suffix) for x in weights]
-                setattr(self, weights[0], w_ih)
+                if not (skip_input and layer == 0):
+                    setattr(self, weights[0], w_ih)
```

Comment:
It's better to use `self.register_parameter` and give it `None` if there's no parameter. Just add an if in the parameters above, that initializes some of them to Nones, and use `register_parameter` instead of assignments. This will also simplify `self._all_weights`, because you'll be able to use regular names, and they will be gathered as `None`.